src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpReceiver.php

```diff
@@ -55,40 +64,70 @@ public function receive(callable $handler): void
             }
 
             try {
-                $handler($this->serializer->decode([
-                    'body' => $AMQPEnvelope->getBody(),
-                    'headers' => $AMQPEnvelope->getHeaders(),
-                ]));
-
-                $this->connection->ack($AMQPEnvelope);
-            } catch (RejectMessageExceptionInterface $e) {
-                try {
-                    $this->connection->reject($AMQPEnvelope);
-                } catch (\AMQPException $exception) {
-                    throw new TransportException($exception->getMessage(), 0, $exception);
-                }
+                $envelope = $this->serializer->decode([
+                    'body' => $this->currentAmqpEnvelope->getBody(),
+                    'headers' => $this->currentAmqpEnvelope->getHeaders(),
+                ]);
+            } catch (MessageDecodingFailedException $exception) {
```

Comment:
This is a responsibility that we *could* move into `Worker`, so that it's not something that the individual receivers needs to worry about.

Reply:
I agree 👍